### PR TITLE
wasi: backfill docs and tests of fd_read and fd_write

### PR DESCRIPTION
**Pull request by nullpo-head**
This PR backfills the docs and tests of `fd_write`, and backfills the tests of `fd_read`.

### CODE REVIEW

**Review by codefromthecrypt — APPROVED** [review]
Thanks, I'm going to add a commit for polish and get this merged!

**Comment by codefromthecrypt** [comment]
polished the docs, will do the unit tests next

**Comment by codefromthecrypt** [comment]
ok done rewriting the tests also

File: internal/wasi/wasi_test.go [hunk 1]
```diff
@@ -185,7 +188,7 @@ func TestAPI_ArgsSizesGet_Errors(t *testing.T) {
 		{
 			name:        "argvBufSize exceeds the maximum valid size by 1",
 			argc:        validAddress,
-			argvBufSize: memorySize - 4 + 1, // 4 is the size of uint32, the type of the buffer size
+			argvBufSize: memorySize - 4 + 1, // 4 is the size of uint32, the type of the length
```
Comment (codefromthecrypt) [inline internal/wasi/wasi_test.go:191]:
```suggestion
			argvBufSize: memorySize - 4 + 1, // 4 the encoded size of uint32le
```

**Comment by codefromthecrypt** [comment]
Since this was a cleanup PR anyway, I'll go ahead and merge my revisions to this. Thanks for the first stab @nullpo-head and looking forward to the next!

**Comment by nullpo-head** [comment]
Thanks for the changes!

**Pull request merged**